cmd/syncthing/support_bundle.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (C) 2014 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package main
+
+import (
+	"archive/zip"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/syncthing/syncthing/lib/config"
+)
+
+// AddBufferFile adds new json to fileByte(name, bytes.Buffer)
+func addBufferFile(arrayFile []fileByte, name string, json []byte) []fileByte {
+	var singleFile fileByte
+	singleFile.name = name
+	singleFile.buf.Write(json)
+	arrayFile = append(arrayFile, singleFile)
+	return arrayFile
+}
+
+// GetRedactedConfig redacting some parts of config
+func getRedactedConfig(s *apiService) config.Configuration {
+	rawConf := s.cfg.RawCopy()
+	rawConf.GUI.APIKey = "REDACTED"
+	rawConf.GUI.Password = "REDACTED"
+	rawConf.GUI.User = "REDACTED"
+	return rawConf
+}
+
+// GetusageReportingData collects needed data for support personnel
+func getusageReportingData(s *apiService, r *http.Request) map[string]interface{} {
```

Comment:
What does this do that we can't just use the return value of `reportData()` directly?

Reply:
Thanks ur comments. I will implement them.